### PR TITLE
Problem: mempool test is not stable

### PR DESCRIPTION
**Pull request by mmsqe**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

close: #649
# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: integration_tests/test_mempool.py [hunk 1]
```diff
@@ -24,50 +26,75 @@ def cronos_mempool(tmp_path_factory):
...
 
-    block_num_1 = w3.eth.get_block_number()
-    assert block_num_1 == block_num_0, "new block is generated, fail"
-    print(f"all send tx hash: f{sended_hash_set}")
+        block_num_1 = w3.eth.get_block_number()
+        try:
+            assert block_num_1 == block_num_0, "new block is generated"
+            print(f"all send tx hash: f{sended_hash_set}")
+        except AssertionError as e:
+            if x < max_retry - 1:
```
Comment (adu-crypto) [inline integration_tests/test_mempool.py:69]: this means at most 2 times retry, is this intended?

File: integration_tests/test_mempool.py [hunk 2]
```diff
@@ -24,50 +26,75 @@ def cronos_mempool(tmp_path_factory):
...
+                print(f"retry {x} failed:", e)
+                continue
```
Comment (adu-crypto) [inline integration_tests/test_mempool.py:71]: the last times exception is caught but not raised.

**Review by adu-crypto — APPROVED** [review]
others LGTM

File: integration_tests/test_mempool.py (hunk 1 above)
Comment (mmsqe) [reply integration_tests/test_mempool.py:69]: 3: 0, 1, 2

File: integration_tests/test_mempool.py [hunk 3]
```diff
@@ -24,50 +26,75 @@ def cronos_mempool(tmp_path_factory):
 
 
 def test_mempool(cronos_mempool):
-    w3: Web3 = cronos_mempool.w3
-    filter = w3.eth.filter("pending")
-    assert filter.get_new_entries() == []
+    max_retry = 3
+    for x in range(max_retry):
```
Comment (yihuang) [inline integration_tests/test_mempool.py:30]: What's the purpose of retry here?

File: integration_tests/test_mempool.py [hunk 4]
```diff
@@ -24,50 +26,75 @@ def cronos_mempool(tmp_path_factory):
...
+        # check after max 10 blocks
+        for i in range(10):
+            all_pending = w3.eth.get_filter_changes(filter.filter_id)
+            print(f"all pending tx hash at block {i+block_num_1}: {all_pending}")
+            for h in all_pending:
+                sended_hash_set.discard(h)
+            if len(sended_hash_set) == 0:
+                break
+
+            # FIXME remove after new design with subscription
```
Comment (yihuang) [inline integration_tests/test_mempool.py:82]: No need to cover up the failure in the test, after all, it's the purpose of the test to reveal the issue.

**Review by yihuang — CHANGES_REQUESTED** [review]
Don't cover up the failure of the filter API by change the test, it defeats the purpose of the test case, right?
we should fix the root cause directly.

File: integration_tests/test_mempool.py (hunk 2 above)
Comment (mmsqe) [reply integration_tests/test_mempool.py:71]: `if x < max_retry - 1` last one is throw

Comment (mmsqe) [reply integration_tests/test_mempool.py:71]: actually we still get_filter_changes even pending txs get confirmed without reconnect issue, not sure if need force `no new block is generated`

File: integration_tests/test_mempool.py (hunk 4 above)
Comment (mmsqe) [reply integration_tests/test_mempool.py:82]: I might misunderstand the purpose of this mempool test, but current behavior seems no patching for txs when [reconnecting](https://github.com/tendermint/tendermint/blob/main/rpc/jsonrpc/client/ws_client.go#L362-L369), so [filter](https://github.com/evmos/ethermint/blob/9bbf356c6b8faca6314ae46f63ae21fcdcd7e866/rpc/namespaces/ethereum/eth/filters/api.go#L167) wont find the missing txs

**Review by yihuang — APPROVED** [review]
LGTM

**Pull request merged**